### PR TITLE
feat(storage): Estimating lsm stat

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Introduce a new algorithm to estimating compact pending bytes which calculates the number of compact bytes needed to balance the LSM Tree from the current state of each level in the LSM Tree in combination with `compaction_config`

support new metrics to estimate compression ratio for each level of lsm

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
~- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features).~
~- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

- [ ] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/meta/src/hummock/metrics_utils.rs [hunk 1]
```diff
@@ -222,3 +226,66 @@ pub fn trigger_safepoint_stat(metrics: &MetaMetrics, safepoints: &[HummockVersio
...
+                match compact_task.compression_algorithm {
+                    1 => "Lz4",
+                    2 => "Zstd",
+                    _ => "",
+                }
+            } else {
+                ""
+            };
+
+            println!(
```
Comment (zwang28) [inline src/meta/src/hummock/metrics_utils.rs:276]: debug info

File: src/storage/src/hummock/compactor/mod.rs [hunk 2]
```diff
@@ -155,9 +155,12 @@ impl Compactor {
 
         let need_quota = estimate_memory_use_for_compaction(&compact_task);
         tracing::info!(
-            "Ready to handle compaction task: {} need memory: {}",
+            "Ready to handle compaction task: {} need memory: {} select_level {} target_level {} compression_algorithm {:?}",
             compact_task.task_id,
-            need_quota
+            need_quota,
+            compact_task.input_ssts[0].level_idx,
+            compact_task.input_ssts[1].level_idx,
```
Comment (zwang28) [inline src/storage/src/hummock/compactor/mod.rs:162]: This implies input_ssts.len() >= 0. But we haven't forced it yet.

File: src/meta/src/hummock/manager/mod.rs [hunk 3]
```diff
@@ -1189,6 +1188,22 @@ where
 
                 trigger_version_stat(&self.metrics, current_version, &versioning.version_stats);
 
+                {
+                    let group_config = self
+                        .get_compaction_group_config(compact_task.compaction_group_id)
+                        .await;
+
+                    trigger_lsm_stat(
```
Comment (hzxa21) [inline src/meta/src/hummock/manager/mod.rs:1196]: `trigger_lsm_stat` is executed when holding the version and compaction lock and we will basically traverse the LSM tree to calculate the pending bytes and compression ratio. Can this be a problem?

File: src/meta/src/hummock/manager/mod.rs [hunk 4]
```diff
@@ -1958,6 +1957,54 @@ where
...
+
+                let id_to_config = hummock_manager.get_compaction_group_map().await;
+                let current_version = {
+                    let mut versioning_guard =
+                        write_lock!(hummock_manager.as_ref(), versioning).await;
+                    versioning_guard.deref_mut().current_version.clone()
+                };
+
+                let compaction_group_ids_from_version = get_compaction_group_ids(&current_version);
+                let default_config = CompactionConfigBuilder::new().build();
```
Comment (zwang28) [inline src/meta/src/hummock/manager/mod.rs:1987]: The `default_config` should be the one from CompactionGroupManager::default_config.

**Review by zwang28 — APPROVED** [review]
rest LGTM

**Pull request merged**